### ISSUE TITLE
Heatmap: Pixelated when zoomed in and offsetted

### ISSUE THREAD

**Issue by janhartmann** [visgl/deck.gl#6221]
#### Description
I am using MapBox together with the HeatmapLayer of deck.gl because I like the the scaling features of the color domain, which compared to the regular MapBox heat-map which does not scale.

I am using the `MapBoxLayer` to seamlessly bind my heat-map layer together with my map. I have frequently updates of the data points (every 500ms). My layer integration is the following:

```ts
import * as React from "react";
import { HeatmapLayer } from "@deck.gl/aggregation-layers";
import { MapboxLayer } from "@deck.gl/mapbox";

export interface IProps {
  data: HeatmapPoint[]; // geojson
}

const HEATMAP_LAYER_ID = "heatmap-deckgl";

const layer = new MapboxLayer({
  id: HEATMAP_LAYER_ID,
  type: HeatmapLayer,
  data: [],
  opacity: 0.9,
  colorRange: [
    [1, 152, 189, 70],
    [73, 227, 206, 120],
    [216, 254, 181, 160],
    [254, 237, 177, 200],
    [254, 173, 84, 210],
    [209, 55, 78, 255]
  ],
  radiusPixels: 30,
  getPosition: (d: HeatmapPoint) => d.geometry.coordinates,
  getWeight: 1,
  threshold: 0.08
});

export const MyHeatmapLayer: React.FC<IProps> = ({ data, children }) => {
  const map = useMap(); // gives an instance of MapBox from a context.

  React.useEffect(() => {
    map.addLayer(layer);

    return () => {
      try {
        map.removeLayer(HEATMAP_LAYER_ID);
      } catch {}
    };
  }, [map]);

  React.useEffect(() => {
    layer.setProps({ data });
  }, [data]);

  return <>{children}</>;
};
```

I am observing that my rending of the heat-map becomes pixelated when I zoom in and out on the map and at certain zoom levels offsets the whole heta-map. If I remove the layer and add it again, it it is again rendered fine. However, zooming shows the pixelation again.

After adding the layer:

<img width="947" alt="Screen Shot 2021-09-22 at 3 35 29 PM" src="https://user-images.githubusercontent.com/1107383/134336572-ae1c400c-d130-4042-942c-5917b8b66e07.png">

Zooming the hot area:

<img width="1333" alt="Screen Shot 2021-09-22 at 3 35 54 PM" src="https://user-images.githubusercontent.com/1107383/134336624-759adfaf-c875-4111-a9d2-501e73d6ee12.png">

#### Expected Behavior
Not pixelate like this. I am unsure what is actually happening. It feels like the layer itself is being zoomed, without adjusting. Like zooming into the layer, like when zooming a static image.

#### Environment
 - Framework Version: "deck.gl": "8.5.9", "mapbox-gl": "2.4.1"
 - Browser Version: Version 93.0.4577.82 (Official Build) (x86_64)
 - OS: Mac OS 10.14.6

**Comment by janhartmann** [comment]
I tried creating this GIF of the problem, notice also the offset at different zoom levels. Low quality, I know. :-)

![demo2](https://user-images.githubusercontent.com/1107383/134489088-dead2bef-8fbb-4377-8be5-168df92049f0.gif)

**Comment by janhartmann** [comment]
Investigating this further, it appears this has something to do with my updates of my data. If I do not update the layer data (except once) the heatmap is drawn correctly and without rendering problems or offset.

**Comment by Pessimistress** [comment]
Thanks for reporting, You are right that this is caused by the cancellation of debounce. When the debounce timer is up the layer performs a full update to the weight map bounds, but fails to do so when debounce is cancelled. Coincidentally your data update interval (500ms) is the same as the debounce, so the denounced code path is never hit.

**Comment by Pessimistress** [comment]
Fixed in 8.5.10

**Comment by janhartmann** [comment]
Worked beautifully, thank you.
